libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -3799,6 +3824,11 @@ static void ProcessSegments(pugi::xml_node Parent,
         /* Get next Node */
         Node = Node.next_sibling(Node.name());
     }
+    /*if no segment with default axis option isn't provided, atleast 1 segment along each x & y axes is needed */
+    if (!both_axis_segs && !(x_axis_segs && y_axis_segs && true)) {
```

Comment:
I think you can simplify this logic, and use more clear variable naming.  This is an x_axis_segs --> x_axis_seg_found (and so on).  Get rid of both_axis_seg and just set x_axis_seg_found = TRUE if an x-segment is found, and both x_axis_seg_found and y_axis_seg_found = TRUE if you have a segment with neither x nor y set.

Reply:
Correct. I updated the logic accordingly.